core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -2317,10 +2318,10 @@ public void finalizeUpdateOfExchangeSinkInstanceHandle(TaskId taskId, ExchangeSi
             partition.updateExchangeSinkInstanceHandle(taskId, updatedExchangeSinkInstanceHandle);
         }
 
-        public void taskFinished(TaskId taskId, TaskStatus taskStatus)
+        public Optional<List<PrioritizedScheduledTask>> taskFinished(TaskId taskId, TaskStatus taskStatus)
```

Comment:
I wouldn't mind seeing a blurb that describes what the return value is supposed to represent.  Something along the lines of
> This method may decide to fail the task, in which case it returns the taskFailed() return value.

Reply:
Will - pass - partly because I am lazy - and partly because it is all enclosed to this class and quite straightforward to read.